src/ml_gen_detection/dagpap24_baseline.py

```diff
@@ -0,0 +1,304 @@
+import argparse
+import gc
+import json
+import logging
+import os
+import random
+from itertools import chain
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import yaml
+from sklearn.model_selection import train_test_split
+from tqdm.auto import tqdm
+
+from src.utils import get_project_root
+
+
+from src.ml_gen_detection.hf_token_classification import main as hf_token_classification
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+"""
+This script performs span prediction and classification with bert models.
+"""
+
+
+def chunk_tokens_labels(df: pd.DataFrame, max_length: int):
+    """
+    This function chunks tokens and their respective labels to
+    max_length token length
+    """
+    index_list = []
+    tokens_list = []
+    labels_list = []
+    for index, row in tqdm(df.iterrows(), total=len(df)):
+        if len(row["token_label_ids"]) > max_length:
+            remaining_tokens = row["tokens"]
+            remaining_labels = row["token_label_ids"]
+
+            # While the remaining list is larger than max_length,
+            # truncate and append
+            while len(remaining_labels) > max_length:
+                index_list.append(index)
+                tokens_list.append(remaining_tokens[:max_length])
+                labels_list.append(remaining_labels[:max_length])
+                remaining_tokens = remaining_tokens[max_length:]
+                remaining_labels = remaining_labels[max_length:]
+            # Append last chunk
+            index_list.append(index)
+            tokens_list.append(remaining_tokens)
+            labels_list.append(remaining_labels)
+        else:
+            index_list.append(index)
+            tokens_list.append(row["tokens"])
+            labels_list.append(row["token_label_ids"])
+
+    return pd.DataFrame(
+        {"index": index_list, "tokens": tokens_list, "labels": labels_list}
+    )
+
+
+def write_df_to_json(df: pd.DataFrame, path_to_json: str):
+    """
+    This function writes pandas dataframes into a compatible json format
+    to be used by huggingface_ner
```

Comment:
```suggestion
    to be used by huggingface_ner
```
to be used by `hf_token_classification`